locked-asset/proxy_dex/src/proxy_pair.rs

```diff
@@ -166,14 +168,19 @@ pub trait ProxyPairModule:
                 .base_asset_token_ref
                 .token_identifier
                 .clone();
-            let unlocked_amount = base_asset_amount_received - locked_token_amount_available;
-            let unlocked_tokens = EsdtTokenPayment::new(asset_token_id.clone(), 0, unlocked_amount);
+            let remaining_locked_amount = base_asset_amount_received - locked_token_amount_available;
+
+            // Lock the difference from IL and send
+            let _ = self.lock_tokens_and_forward(
+                caller.clone(),
+                EgldOrEsdtTokenIdentifier::esdt(asset_token_id.clone()),
+                remaining_locked_amount,
+            );
 
             // burn base asset, as we only need to send the locked tokens
             self.send()
```

Comment:
You should only burn `attributes.locked_tokens.amount - remaining_locked_amount`, since `remaining_locked_amount` is sent to the energy factory for locking.